iree/compiler/Dialect/Flow/Transforms/FoldCompatibleDispatchRegions.cpp

```diff
@@ -363,7 +394,7 @@ LogicalResult mergeBlockDispatchRegions(FuncOp func, Block *parentBlock) {
         LLVM_DEBUG(llvm::dbgs()
                    << "   -REGION CONTAINS NON-TRIVIAL CONTROL FLOW-\n");
       }
-      if (rhsHasRootOnlyOp(lhs, rhs)) {
+      if (rhsHasRootOnlyOp(lhs, rhs) || lhsHasLeafOnlyOp(lhs, rhs)) {
```

Comment:
This is a bit confusing. The logic of `rhsHasRootOnlyOp` seems to be that if the `rhs` is a root op then dont merge the dispatch region. But if lhs is `leaf` op you can still merge the dispatch region....

Reply:
I think the meaning of RootOp and LeafOp is unclear here as well, so it looks confusing.

Does it make sense if we see these as `rhsHasFirstOnlyOp` and `lhsHasLastOnlyOp`?